OADP-6290: Created seperate modules from the OADP API assembly [mod-docs work]

apurvabhide17 opened the pull request:
Version(s):
OCP 4.20

Issue:
[OADP-6290](https://issues.redhat.com/browse/OADP-6290)

Link to docs preview: https://94973--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-api.html


QE review:
- [ ] QE has approved this change.

Note - 
- Modularization work for the oadp-api assembly. Instead of having multiple tables in the assembly, I've moved the content to new modules. Otherwise, there are no technical changes.
- As discussed with Kathryn, skipping the QE review because this PR focuses on changing structure and there are no technical changes.

apurvabhide17 committed:
OADP-6290: Created seperate modules from the OADP API assembly

Fixing xref issue

apurvabhide17 commented:
/label OADP

anarnold97 commented on modules/podconfig-type.adoc:
```diff
@@ -0,0 +1,36 @@
+|`nodeSelector`
+|map [ link:https://pkg.go.dev/builtin#string[string] ] link:https://pkg.go.dev/builtin#string[string]
+|Defines the `nodeSelector` to be supplied to a `Velero` `podSpec` or a `Restic` `podSpec`.
+For more details, see xref:../../backup_and_restore/application_backup_and_restore/oadp-api.adoc#oadp-configuring-node-agents_oadp-api[Configuring node agents and node labels].
```
@apurvabhide17 - you have a Vale error, here

what do you think is the best resolution? will you add an `.Additional resources` section in the assembly immediately following the `include` for this module?

apurvabhide17 replied:
done, I've fixed it

anarnold97 commented on modules/podconfig-type.adoc:
```diff
@@ -0,0 +1,36 @@
+
+|`resourceAllocations`
+|link:https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements[ResourceRequirements]
+|Set specific resource `limits` and `requests` for a `Velero` pod or a `Restic` pod as described in xref:../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc#oadp-setting-resource-limits-and-requests_installing-oadp-aws[Setting Velero CPU and memory resource allocations].
```
@apurvabhide17 - you have a Vale error, here

what do you think is the best resolution? will you add an `.Additional resources` section in the assembly immediately following the `include` for this module? 

i think this might be the easiest and best solution

anarnold97 reviewed (changes requested):
LGTM - apart from the `xrefs` in modules, which is throwing errors in the Vale bot

not sure what the best solution is. 

please can you have a think about it and talk to the team

thanks

apurvabhide17 commented:
/label peer-review-needed

apurvabhide17 commented:
Hi @dfitzmau, done, I've squashed the commits.

dfitzmau reviewed (approved):
Nice work. I added comments inline. I think the VeleroConfig type file needs an include?

apurvabhide17 commented:
> Nice work. I added comments inline. I think the VeleroConfig type file needs an include?

@dfitzmau, ack, thanks. I've implemented your suggestions. 
Now, adding merge-review label.

kcarmichael08 commented on backup_and_restore/application_backup_and_restore/oadp-api.adoc:
```diff
@@ -9,246 +9,52 @@ include::_attributes/common-attributes.adoc[]
+* `Features`
+* `DataMover`
++
+For more information, see in link:https://pkg.go.dev/github.com/openshift/oadp-operator[OADP Operator](Go documentation).
```
```suggestion
For more information, see in link:https://pkg.go.dev/github.com/openshift/oadp-operator[OADP Operator] (Go documentation).
```
I won't hold up merging this but if you make changes, might want to fix this.

kcarmichael08 commented on backup_and_restore/application_backup_and_restore/oadp-api.adoc:
```diff
@@ -9,246 +9,52 @@ include::_attributes/common-attributes.adoc[]
+.Additional resources
+* xref:../../backup_and_restore/application_backup_and_restore/oadp-api.adoc#oadp-configuring-node-agents_oadp-api[Configuring node agents and node labels]
+* xref:../../backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc#oadp-setting-resource-limits-and-requests_installing-oadp-aws[Setting Velero CPU and memory resource allocations]
+* link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig[Complete schema definitions for the type `PodConfig`](Go documentation)
```
```suggestion
* link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig[Complete schema definitions for the type `PodConfig`] (Go documentation)
```
